lib/sign_notarize.sh

```diff
@@ -0,0 +1,105 @@
+if [[ $UNAME == Darwin ]]; then
+  if [[ -n "$NODMG" ]]; then
+    echo "Skipping Notarization and DMG generation."
+    exit 0
+  fi
+
+  # Prepare
+  rm -rf *.dmg
+  DMG_NOTARIZE_BASE="${WORKSPACE}/dmg_notarize_base"
+
+  # JDK
+  if [ "$RELEASE_BUILD" == true ]; then
+    security unlock-keychain -p $unlockpass ~/Library/Keychains/login.keychain
+    xcrun notarytool submit --force --keychain-profile "sapmachine-notarization" --wait "${WORKSPACE}/${ARCHIVE_NAME_JDK}"
+  fi
+  DMG_NAME_JDK=$(basename ${ARCHIVE_NAME_JDK} .tar.gz)
+  rm -rf ${DMG_NOTARIZE_BASE}
+  mkdir -p ${DMG_NOTARIZE_BASE}
+  tar -xzf "${WORKSPACE}/${ARCHIVE_NAME_JDK}" -C ${DMG_NOTARIZE_BASE}
+  hdierror=0
+  hdiutil create -verbose -srcfolder ${DMG_NOTARIZE_BASE} -fs HFS+ -volname ${DMG_NAME_JDK} "${WORKSPACE}/${DMG_NAME_JDK}.dmg" || hdierror=1
+  if [ $hdierror -ne 0 ]; then
+    # We see sometimes errors like "hdiutil: create failed - Resource busy." when invoking it right after tar.
+    # Let's retry after sleeping a little while.
+    sleep 30
+    hdiutil create -verbose -srcfolder ${DMG_NOTARIZE_BASE} -fs HFS+ -volname ${DMG_NAME_JDK} "${WORKSPACE}/${DMG_NAME_JDK}.dmg"
+  fi
+  echo "${DMG_NAME_JDK}.dmg" > "${WORKSPACE}/jdk_dmg_name.txt"
+
+  if [ "$RELEASE_BUILD" == true ]; then
+    xcrun stapler staple ${DMG_NOTARIZE_BASE}/*
+    rm "${WORKSPACE}/${ARCHIVE_NAME_JDK}"
+    tar -czf "${WORKSPACE}/${ARCHIVE_NAME_JDK}" -C ${DMG_NOTARIZE_BASE} .
+    xcrun notarytool submit --keychain-profile "sapmachine-notarization" --wait "${WORKSPACE}/${DMG_NAME_JDK}.dmg"
+    xcrun stapler staple "${WORKSPACE}/${DMG_NAME_JDK}.dmg"
+  fi
+
+  # JRE
+  if [ "$RELEASE_BUILD" == true ]; then
+    xcrun notarytool submit --force --keychain-profile "sapmachine-notarization" --wait "${WORKSPACE}/$ARCHIVE_NAME_JRE"
+  fi
+  DMG_NAME_JRE=$(basename ${ARCHIVE_NAME_JRE} .tar.gz)
+  rm -rf ${DMG_NOTARIZE_BASE}
+  mkdir -p ${DMG_NOTARIZE_BASE}
+  tar -xzf "${WORKSPACE}/${ARCHIVE_NAME_JRE}" -C ${DMG_NOTARIZE_BASE}
+  hdierror=0
+  hdiutil create -verbose -srcfolder ${DMG_NOTARIZE_BASE} -fs HFS+ -volname ${DMG_NAME_JRE} "${WORKSPACE}/${DMG_NAME_JRE}.dmg" || hdierror=1
+  if [ $hdierror -ne 0 ]; then
+    # We see sometimes errors like "hdiutil: create failed - Resource busy." when invoking it right after tar.
+    # Let's retry after sleeping a little while.
+    sleep 30
+    hdiutil create -verbose -srcfolder ${DMG_NOTARIZE_BASE} -fs HFS+ -volname ${DMG_NAME_JRE} "${WORKSPACE}/${DMG_NAME_JRE}.dmg"
+  fi
+  echo "${DMG_NAME_JRE}.dmg" > "${WORKSPACE}/jre_dmg_name.txt"
+
+  # Notarize if doing a release build
+  if [ "$RELEASE_BUILD" == true ]; then
+    xcrun stapler staple ${DMG_NOTARIZE_BASE}/*
+    rm "${WORKSPACE}/${ARCHIVE_NAME_JRE}"
+    tar -czf "${WORKSPACE}/${ARCHIVE_NAME_JRE}" -C ${DMG_NOTARIZE_BASE} .
+    xcrun notarytool submit --keychain-profile "sapmachine-notarization" --wait "${WORKSPACE}/${DMG_NAME_JRE}.dmg"
+    xcrun stapler staple "${WORKSPACE}/${DMG_NAME_JRE}.dmg"
+  fi
+fi
```

Comment:
linebreak at the end is missing (convention)

Reply:
done